tests/registration_test.py

```diff
@@ -677,14 +677,13 @@ def test_segarray_register_attach(self):
         a = [1, 2, 3]
         b = [6, 7, 8]
 
-        segarr = ak.segarray(ak.array([0, len(a)]), ak.array(a + b))
+        segarr = ak.SegArray(ak.array([0, len(a)]), ak.array(a + b))
         # register the seg array
         segarr.register("segarrtest")
         ak.clear()
         segarr_2 = ak.SegArray.attach("segarrtest")
 
         self.assertEqual(segarr.size, segarr_2.size)
-        self.assertListEqual(segarr.lengths.to_list(), segarr_2.lengths.to_list())
```

Comment:
i feel like this test would is still useful and we just recalculate `lengths` if needed